[3890] Update the Bulk Upload function to support larger amounts of files and be more precise

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/3890

aachavarria committed:
adding DialogTitle.tsx;
adding getBulkUploadUrl to content.ts;
adding hooks.ts;
adding string.ts;
adding BulkUpload with custom DropZone;

rart commented on ui/app/src/components/BulkUpload.tsx:
```diff
@@ -0,0 +1,471 @@
+  const {
+    onClose = () => {
+    },
+    path = '/static-assets/test',
```
Fix

rart commented on ui/app/src/components/BulkUpload.tsx:
```diff
@@ -0,0 +1,471 @@
+    onClose = () => {
+    },
+    path = '/static-assets/test',
+    site = 'editorial',
```
Fix

rart commented on ui/app/src/components/BulkUpload.tsx:
```diff
@@ -0,0 +1,471 @@
+    },
+    path = '/static-assets/test',
+    site = 'editorial',
+    limit = 2
```
Rename to maxSimultaneousUploads, default to 1

rart commented on ui/app/src/components/BulkUpload.tsx:
```diff
@@ -0,0 +1,471 @@
+
+  const onUpload = () => {
+    setStatus('uploading');
+  };
```
Think it makes sense to have a single prop for this: `onStatusChange`

rart commented on ui/app/src/components/BulkUpload.tsx:
```diff
@@ -0,0 +1,471 @@
+      <path d="M11 10V0H5v10H2l6 6 6-6h-3zm0 0" fillRule="evenodd"/>
+    </svg>
+  )
+}
```
- Format nicely
- Rename to ArrowPointingDown

aachavarria replied:
removed

rart commented on ui/app/src/components/BulkUpload.tsx:
```diff
@@ -0,0 +1,471 @@
+          target: generalProgress.current,
+          hideAfterFinish: false
+        })
+        .setMeta({ site });
```
Format

rart commented on ui/app/src/components/DialogTitle.tsx:
```diff
@@ -0,0 +1,71 @@
+/*
```
All the annotations done on the Site Translation / Pages widgets PR for this component apply here.

rart commented on ui/app/src/utils/string.ts:
```diff
@@ -40,6 +40,14 @@ export function isBlank(str: string): boolean {
   return str === '';
 }
 
+export function bytesToSize(bytes: number, seperator: string = '') {
```
```ts
export function bytesToSize(bytes: string, separator: string = '');
export function bytesToSize(bytes: number, separator: string = '');
export function bytesToSize(bytes: number | string, separator: string = '') {
  ...
```
(notice typo in the word separator)

Add to default exports

aachavarria committed:
styles fixes

aachavarria committed:
fileInput and thumbGeneration

aachavarria committed:
adding browse btn;
adding forwardRef;
adding validation to duplicates files;

aachavarria committed:
bytesToSize renaming 'separator';
adding rounded to DialogTitle;

aachavarria committed:
DialogTitle update;
BulkUpload: adding minimized; changes on reporting Status;

aachavarria committed:
adding uploading status;

aachavarria committed:
deleting console.log

aachavarria committed:
wcm-assets-folder.js: adding validation on bulkUpload;
confirmDialog.tsx: changes to support onCancel, onOk;
codebase-bridge.tsx: adding ConfirmDialog;
i18n: adding bulkUploadConfirmDialogMessages;

aachavarria committed:
onbeforeunload changes;

aachavarria committed:
adding Refresh();

aachavarria commented on ui/app/src/components/DialogTitle.tsx:
```diff
@@ -0,0 +1,71 @@
+  onClose(): void;
+}
+
+export default function DialogTitle(props: DialogTitleProps) {
```
Change the name to DialogHeader

aachavarria commented on ui/app/src/components/DialogTitle.tsx:
```diff
@@ -0,0 +1,71 @@
+  return (
+    <MuiDialogTitle disableTypography className={classes.titleRoot}>
+      <div className={classes.title}>
+        <Typography variant="h6">{title}</Typography>
```
add props (with default values) for titleTypographyProps, subtitleTypographyProps. Put variants in there

aachavarria committed:
reniaming dialogTitle to dialogHeader

aachavarria committed:
removing ts-ignore

aachavarria committed:
Deleting commentes

aachavarria committed:
BulkUpload: Changing upload messages;
Code format;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3890

aachavarria committed:
maxSimultaneousUploads to 2
